tests/main/test_openai_wrapper.py

```diff
@@ -113,6 +75,282 @@ def test_predict_call(self):
                     'total_tokens': expected_total_tokens
                 }
             }
+        return mock_result
+
+    @pytest.mark.skipif(sys.version_info.minor <= 6,
+                        reason='Openai not supported for older versions')
+    def test_predict_call(self):
+        # test creating the OpenaiWrapperModel and
+        # calling the predict function
+        api_type = 'azure'
+        api_base = 'https://mock.openai.azure.com/'
+        api_version = '2023-03-15-preview'
+        api_key = 'mock'
+        context = ''
+        questions = 'How to convert 10^9/l to liter?'
+        is_openai_1_0 = False
+        if not hasattr(openai, 'OpenAI'):
+            mock_function = 'openai.ChatCompletion.create'
+        else:
+            mock_function = 'openai.resources.chat.completions.Completions.create'
+            is_openai_1_0 = True
+        answer = (' It seems there is some confusion with the units being'
```

Comment:
a lot of the code, especially the text, is duplicated across multiple test cases - we should refactor it for easier maintenance